src/robot/running/arguments/typeinfo.py

```diff
@@ -276,6 +280,37 @@ def from_sequence(cls, sequence: 'tuple|list') -> 'TypeInfo':
             return infos[0]
         return cls('Union', nested=infos)
 
+    @classmethod
+    def from_variable(cls, variable: 'str|VariableMatch',
+                      handle_list_and_dict: bool = True) -> 'TypeInfo|None':
+        """Construct a ``TypeInfo`` based on a variable."""
+        if isinstance(variable, str):
+            variable = search_variable(variable, parse_type=True)
```

Comment:
It would be a little bit more clear to change the condition to `if not isinstance(variable, VariableMatch)`. It's not obvious to known that `search_variable` returns `VariableMatch` by looking just the code.